support-data-rest/src/main/java/org/uniprot/api/literature/service/LiteratureService.java

```diff
@@ -68,6 +54,7 @@ public LiteratureEntry findById(final String literatureId) {
                 SolrRequest.builder()
                         .query("mapped_protein:" + accession)
                         .addSort(literatureSortClause.getSort(requestDTO.getSort(), false))
+                        .rows(BasicSearchService.DEFAULT_SOLR_BATCH_SIZE)
```

Comment:
I might be wrong, but I think this should take the request.getSize into account, right? because this way the page size will always be 25.

Reply:
Fixed it. See the code.